src/shogun/labels/BinaryLabels.cpp

```diff
@@ -139,8 +141,28 @@ CLabels* CBinaryLabels::duplicate() const
 	return new CBinaryLabels(*this);
 }
 
-namespace shogun
-{
+
+namespace shogun {
+	SG_FORCED_INLINE Some<CBinaryLabels> from_multiclass(CMulticlassLabels* orig) {
+		auto result_vector = orig->get_labels();
+		std::set<int32_t> unique(result_vector.begin(), result_vector.end());
+		auto min = *std::min_element(unique.begin(), unique.end());
+
+		ASSERT(unique.size() == 2);
+
+		SGVector<int32_t> converted(result_vector.vlen);
+		std::transform(
```

Comment:
I wonder whether we *always* should print a warning with the conversion table in case we convert (especially if we change the representation). As doing it implicit might lead to confusion. Thoughts?

Reply:
I think we need something like that. Because ::apply will return labels with either -1/+1, which is probably not what the user wanted... But then warnings can be easily missed or be too verbose if you do something like a gridsearch

Reply:
Actually, technically we only need one unique label. For training that never makes sense. But I think evaluation methods might complain since they also convert. Think test labels all are positive (totally reasonable). So this needs a modification.

Reply:
Ok so this brings up another issue: What if the training labels are somewhat different than the testing labels...will the conversions be the same? If not, what do we do?

Reply:
I think the grid-search thing would need to convert only once at the beginning anyways as otherwise  things will be inefficient ... and also might be buggy (see comment on conversion mapping invariance). So for those one off cases, I would rather warn the user or?

Reply:
OK, I will add a warning. I think in openml this will generate lots of warnings when they do the learning for each fold, but I can disable warnings there. I will add the warning with a mapping.

Reply:
Mmmh
I wonder whether we could do the conversion on the base class just like we did the dense feature type dispatcher.
Because you are of course right

Reply:
I think this would be best, and would avoid any confusions

Reply:
ok, ideas how to get going with that?